orangecontrib/text/widgets/owwordlist.py

```diff
@@ -0,0 +1,605 @@
+    INTERSECT, UNION, INPUT, LIBRARY = range(4)
+    ITEMS = ["Intersection", "Union", "Only Input", "Ignore Input"]
+
+    @staticmethod
+    def update(model: PyListModel, lib_words: List, in_words: List, rule: int):
+        if rule == UpdateRules.INTERSECT:
+            intersect = set(lib_words) & set(in_words)
+            model.wrap(list({k: None for k in lib_words if k in intersect}))
+        elif rule == UpdateRules.UNION:
+            model.wrap(list({k: None for k in lib_words + in_words}))
+        elif rule == UpdateRules.INPUT:
+            model.wrap(list(in_words))
+        elif rule == UpdateRules.LIBRARY:
+            model.wrap(list(lib_words))
+        else:
+            raise NotImplementedError
+
+
+class WordListItemDelegate(QStyledItemDelegate):
+    def displayText(self, word_list: WordList, _) -> str:
+        return "*" + word_list.name if word_list.flags & WordList.Modified \
+            else word_list.name
+
+    def paint(self, painter: QPainter, option: QStyleOptionViewItem,
+              index: QModelIndex):
+        word_list = index.data(Qt.DisplayRole)
+        if word_list.flags & WordList.Modified:
+            option = QStyleOptionViewItem(option)
+            option.palette.setColor(QPalette.Text, QColor(Qt.red))
+            option.palette.setColor(QPalette.Highlight, QColor(Qt.darkRed))
+            option.palette.setColor(QPalette.HighlightedText, QColor(Qt.white))
+        super().paint(painter, option, index)
+
+    def createEditor(self, parent: QWidget, _, __) -> QLineEdit:
+        return QLineEdit(parent)
+
+    def setEditorData(self, editor: QLineEdit, index: QModelIndex):
+        word_list = index.data(Qt.DisplayRole)
+        editor.setText(word_list.name)
+
+    def setModelData(self, editor: QLineEdit,
+                     model: PyListModel, index: QModelIndex):
+        model[index.row()].name = str(editor.text())
+
+
+class ListView(ListViewSearch):
+    drop_finished = Signal()
+
+    def __init__(self):
+        super().__init__(
+            editTriggers=QListView.DoubleClicked | QListView.EditKeyPressed,
+            sizePolicy=QSizePolicy(QSizePolicy.Ignored, QSizePolicy.Expanding),
+            minimumWidth=200,
+            selectionMode=QListView.ExtendedSelection,
+            dragEnabled=True,
+            acceptDrops=True,
+            defaultDropAction=Qt.MoveAction,
+        )
+        self.setDropIndicatorShown(True)
+
+    def startDrag(self, actions):
+        super().startDrag(actions)
+        self.drop_finished.emit()
+
+
+class OWWordList(OWWidget):
+    name = "Word List"
+    description = "Create a list of words."
+    icon = "icons/WordList.svg"
+    priority = 1000
+
+    class Inputs:
+        words = Input("Words", Table)
+
+    class Outputs:
+        selected_words = Output("Selected Words", Table)
+        words = Output("Words", Table)
+
+    class Warning(OWWidget.Warning):
+        no_string_vars = Msg("Input 'Words' needs at least one Text variable.")
```

Comment:
I would omit "Words" as this is the only input there is. Unless we plan to add more than one input?

Reply:
Fixed